How often does the fighter with a higher rank win?  
NB! In each weight class there is a champion and then come fighters ranked from 1 to 15. This means that the fighter who is ranked 1 is not a champion, but number 1 contender for the championship fight (eventhough sometimes lower ranked fighters (2-6) get the opportunity instead - depends on the circumstances)

In [2]:
import pandas as pd

In [3]:
data = pd.read_csv('../../data/ufc-master.csv')

In [4]:
#Filter out fights where neither of the fighters is a champion or ranked in the top 15 because we can't compare the ranks of the
#fightetrs in that case.
ranked_only = data['better_rank'] != 'neither'
ranked_fights = data[ranked_only][['Winner', 'better_rank']]

In [5]:
#Percetange of fights, where the higher ranked fighter wins
better_wins = 0
for i in range(len(ranked_fights)):
    if (ranked_fights['Winner'].iloc[i] == 'Blue') & (ranked_fights['better_rank'].iloc[i] == 'Blue'):
        better_wins += 1
    elif (ranked_fights['Winner'].iloc[i] == 'Red') & (ranked_fights['better_rank'].iloc[i] == 'Red'):
        better_wins += 1
better_wins_p = better_wins / len(ranked_fights)
print("Higher ranked fighters win " + str(round(better_wins_p * 100, 2)) + "% of the time.")

Higher ranked fighters win 60.93% of the time.


This shows that the smart move is to pick the higher ranked fighter if we do not know anything else about both fighters. This is not a suprising result and shows that rankings exist for a reason even though they can be quite arbitrary. For example sometimes a really skilled fighter has a low ranking (10-15) because he/she has difficulties finding higher ranked opponents and the best way to climb in the rankings is to beat fighters that are ranked higher than you.


Using conditional probability let's see how often does the better ranked fighter win, when we know that the fight is a title fight. Basically in most of the cases we are asking how successfully champions defend their champion status.

In [15]:
title_only = (data['title_bout'] == True) & (data['better_rank'] != 'neither')
title_fights = data[title_only][['B_fighter', 'R_fighter', 'Winner', 'better_rank']]

In [16]:
better_wins2 = 0
for i in range(len(title_fights)):
    if (title_fights['Winner'].iloc[i] == 'Red') & (title_fights['better_rank'].iloc[i] == 'Red'):
        better_wins2 += 1
    elif (title_fights['Winner'].iloc[i] == 'Blue') & (title_fights['better_rank'].iloc[i] == 'Blue'):
        better_wins2 += 1
better_wins2_p = better_wins2 / len(title_fights)
print("In the title fights the better ranked fighter wins " + str(round(better_wins2_p * 100, 2)) + "% of the time.")

In the title fights the better ranked fighter wins 70.92% of the time.


71% win rate is significant, which means that it is usually not a good idea to bet against a standing champion or in the case of vacant title it is usually wiser to pick the higher ranked fighter as the next champion.